common/djangoapps/util/model_utils.py

```diff
@@ -0,0 +1,101 @@
+"""
+Utilities for django models.
+"""
+from eventtracking import tracker
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.db.models.fields.related import RelatedField
+
+from django_countries.fields import Country
+
+
+def get_changed_fields_dict(instance, model_class):
+    """
+    Helper method for tracking field changes on a model.
+
+    Given a model instance and class, return a dict whose keys are that
+    instance's fields which differ from the last saved ones and whose values
+    are the old values of those fields.  Related fields are not considered.
+
+    Args:
+        instance (Model instance): the model instance with changes that are
+            being tracked
+        model_class (Model class): the class of the model instance we are
+            tracking
+
+    Returns:
+        dict: a mapping of field names to current database values of those
+            fields
+    """
+    try:
+        old_model = model_class.objects.get(pk=instance.pk)
+    except model_class.DoesNotExist:
+        # Object is new, so fields haven't technically changed.  We'll return
+        # an empty dict as a default value.
+        return {}
+    else:
+        # Don't consider related fields
+        field_names = [
+            field[0].name for field in model_class._meta.get_fields_with_model()
+            if not isinstance(field[0], RelatedField)
+        ]
+        changed_fields = {
+            field_name: getattr(old_model, field_name) for field_name in field_names
+            if getattr(old_model, field_name) != getattr(instance, field_name)
+        }
+
+        return changed_fields
+
+
+def emit_field_changed_events(instance, user, event_name, db_table, excluded_fields=None, hidden_fields=None):
```

Comment:
I would expect some unit tests for this method, specifically around firing multiple events for multiple changed fields, and the behavior of deleting _changed_fields. And the excluded_fields, which I don't think are covered elsewhere.